src/inmanta/server/agentmanager.py

```diff
@@ -495,18 +495,18 @@ async def _log_session_expiry_to_db(
         """
         async with data.AgentProcess.get_connection() as connection:
             async with connection.transaction():
-                await data.Agent.update_primary(tid, endpoints_with_new_primary, now, connection)
```

Comment:
The lock order is documented here:
https://github.com/inmanta/inmanta-core/blob/master/src/inmanta/data/__init__.py#L90

this is also affected by cascading deletes. 

Could you refer to the comment there?